pkg/apis/hive/v1/baremetal/machinepool.go

```diff
@@ -0,0 +1,6 @@
+package baremetal
+
+// MachinePoolPlatform stores the configuration for a machine pool
+// installed on bare metal.
+type MachinePoolPlatform struct {
```

Comment:
If the MachinePool is related to a MachineSet, does this need to have fields like the provider spec for a machine set? The fields for bare metal are in https://github.com/openshift/cluster-api-provider-baremetal/blob/master/pkg/apis/baremetal/v1alpha1/baremetalmachineproviderspec_types.go#L33

Reply:
The way things currently stand with MachinePools is that one MachinePool can represent multiple MachineSets. So a 1-to-many relationship between MachinePools->MachineSets (although you can definitely define a MachinePool with a list of AvailabilityZones of exactly one AZ, and that will result in one single MachineSet being generated).

Reply:
Remote machine pool generation (day 2) should be disabled by default until we add support, so we can add this as followup.